### PR TITLE
Decrease the McCabe complexity of plot_estimate

### PR DESCRIPTION
**Pull request by usmanatron**
Fix for Issue #560 (specifically `mccabe: MC0001 / plot_estimate is too complex (12)`)

Breaking these up have been harder than I hoped :-/.  That said, Ive managed to simplify plot_estimate!
I've done this by adding a "Config" class which takes in the inputs that need pre-processing and does that separately.  Not all the inputs need this, so it's not as awful as it could've been.

Note that I haven't tested this as I don't really understand how to use lifelines.  If you're able to easily manually test that would be awesome, otherwise I can have a go (any examples would be appreciated)

### CODE REVIEW

File: lifelines/plotting.py [hunk 1]
```diff
@@ -332,78 +332,102 @@ def plot_estimate(
...
+
+        if censor_styles is None:
+            self.censor_styles = {}
+
+        if (self.loc is not None) and (self.iloc is not None):
+            raise ValueError("Cannot set both loc and iloc in call to .plot().")
+
+        set_kwargs_ax(kwargs)
+        set_kwargs_color(kwargs)
+        set_kwargs_drawstyle(kwargs)
```
Comment (CamDavidsonPilon) [inline lifelines/plotting.py]: I know a better pattern for these, but I'll change them after this merges.

**Comment by CamDavidsonPilon** [comment]
`plot_estimate` is a hairy beast, written back when I thought I was good at software engineering haha. Overall, this abstraction looks good to me - it's a nice separation of concerns. Thanks @usmanatron! Merging into 0.16.0 now.

File: lifelines/plotting.py [hunk 2]
```diff
@@ -332,78 +332,102 @@ def plot_estimate(
...
-        if bandwidth is None:
-            raise ValueError("Must specify a bandwidth parameter in the call to plot_hazard.")
-        estimate_ = cls.smoothed_hazard_(bandwidth)
-        confidence_interval_ = cls.smoothed_hazard_confidence_intervals_(bandwidth, hazard_=estimate_.values[:, 0])
-    else:
-        estimate_ = getattr(cls, estimate)
-        confidence_interval_ = getattr(cls, "confidence_interval_")
-
+    config = PlotEstimateConfig(
+        cls,
```
Comment (Kilo59) [inline lifelines/plotting.py]: I would avoid using this as a variable name especially in the context of a class method. 
https://stackoverflow.com/a/7554899/6304433

Comment (usmanatron) [reply lifelines/plotting.py]: Good point, Ill fix this and create a new PR

**Pull request merged**